src/common/framework/tree.ts

```diff
@@ -62,30 +62,37 @@ export interface TestTreeLeaf {
 
 export type TestTreeNode = TestSubtree | TestTreeLeaf;
 
+export type ExpandThroughLevel = 1 | 2 | 3;
+
 export class TestTree {
+  readonly forQuery: TestQuery;
```

Comment:
nit: comment what forQuery does - from what I understand, makes it so that iterating the tree filters by the query